src/FSharp.FGL/Undirected.fs

```diff
@@ -115,8 +115,26 @@ module Vertices =
             Map.tryFind v1 g
             |> Option.bind (fun (_, _, s) -> Map.tryFind v2 s)
             |> Option.map (fun e -> (v1,v2,e))
-
-
+    
+    ///Creates a list of all edges and their labels.
+    let toEdgeList (g:Graph<'Vertex,'Label,'Edge>) : LEdge<'Vertex,'Edge> list = 
+        let sourceList = 
+           g
+           |> Graph.mapContexts (fun (p,v,l,s) -> s)
+           |> Map.toList
+           |> List.collect (fun (v,es) -> 
+                es
+                |> List.map (fun (v2,e) -> v,v2,e))
+        let rec takeOnlyOne (i:int) (targetList:LEdge<'Vertex,'Edge>list) =
```

Comment:
Don't iteratively use `List.item` as this is very ineffective. Either use an array, the `::` operator or built in list functions